resources/weather.ts

```diff
@@ -1,24 +1,26 @@
-import gWeatherRates from './weather_rate';
+import WeatherRate from './weather_rate';
 import ZoneInfo from './zone_info';
 
-export function getWeather(timeMs, zoneId) {
+export const getWeather = (timeMs: number, zoneId: number): string | undefined => {
   const chance = getWeatherChanceValue(timeMs);
 
   // See weather_rate.js and territory_type.js for details.
-  const rateIdx = ZoneInfo[zoneId].weatherRate;
-  const entry = gWeatherRates[rateIdx];
+  const rateIdx = ZoneInfo[zoneId]?.weatherRate;
```

Comment:
Should the comment above this be updated?

Reply:
Good idea, thanks!